src/servers/src/system_metric.rs

```diff
@@ -0,0 +1,134 @@
+    config: SystemMetricOption,
+    filter: Option<MetricFilter>,
+}
+
+impl SystemMetricTask {
+    pub fn try_new(config: &SystemMetricOption, plugins: Option<&Plugins>) -> Result<Option<Self>> {
+        if !config.enable {
+            return Ok(None);
+        }
+        let filter = plugins.map(|p| p.get::<MetricFilter>()).unwrap_or(None);
+        ensure!(
+            config.write_interval.as_secs() != 0,
+            InvalidRemoteWriteConfigSnafu {
+                msg: "Expected System metric write_interval greater than zero"
+            }
+        );
+        ensure!(
+            !config.db.is_empty(),
+            InvalidRemoteWriteConfigSnafu {
+                msg: "Expected System metric db not empty"
+            }
+        );
+        Ok(Some(Self {
+            config: config.clone(),
+            filter,
+        }))
+    }
+    pub fn start(&self) {
+        if !self.config.enable {
+            return;
+        }
+        let mut interval = time::interval(self.config.write_interval);
+        let sec = self.config.write_interval.as_secs();
+        let endpoint = format!(
+            "http://{}/v1/prometheus/write?db={}",
+            self.config.endpoint, self.config.db
+        );
+        let filter = self.filter.clone();
+        let _handle = common_runtime::spawn_bg(async move {
+            info!(
+                "Start system metric task to endpoint: {}, interval: {}s",
+                endpoint, sec
+            );
+            // Pass the first tick. Because the first tick completes immediately.
+            interval.tick().await;
+            loop {
+                interval.tick().await;
+                match write_system_metric(&endpoint, filter.as_ref()).await {
+                    Ok(resp) => {
+                        if !resp.status().is_success() {
+                            error!("report system metric error, msg: {:#?}", resp);
+                        }
+                    }
+                    Err(e) => error!("report system metric failed, error {}", e),
+                };
+            }
+        });
+    }
+}
+
+/// Export the collected metrics, encode metrics into [RemoteWrite format](https://prometheus.io/docs/concepts/remote_write_spec/),
+/// and send metrics to Prometheus remote-write compatible receiver (e.g. `greptimedb`) specified by `url`.
+/// User could use `MetricFilter` to filter metric they don't want collect
+pub async fn write_system_metric(url: &str, filter: Option<&MetricFilter>) -> Result<Response> {
+    let metric_families = prometheus::gather();
+    let request = convert_metric_to_write_request(
+        metric_families,
+        filter,
+        Timestamp::current_millis().value(),
+    );
+    // RemoteWrite format require compress by snappy
+    let content = snappy_compress(&request.encode_to_vec())?;
+    let client = reqwest::Client::new();
+    client
+        .post(url)
+        .header("X-Prometheus-Remote-Write-Version", "0.1.0")
```

Comment:
Looks like we don't consider the authentication.

https://docs.greptime.com/user-guide/clients/http-api#authentication

Reply:
But according to the current results, the writing can be successful even without adding authentication parameters. I don’t know if there is an authentication switch or whether the authentication is activate. Maybe we can add some username and password to the configuration option for future compatibility.

Reply:
Another header field will be pass in `[export_metrics]`

Reply:
Yes, the authentication must be configured first, see https://docs.greptime.com/user-guide/clients/authentication

Reply:
`header` has been add as a config option